### ISSUE TITLE
[code] allow to switch between forks

### ISSUE THREAD

**Issue by frank-bee** [gitpod-io/gitpod#4588]
As a github contributor of an upstream project I want to create a private fork and create PRs from there.

Currently this workflow is poorly supported by gitpod.
Why?
Either I launch my gitpod workspace from the upstream project but I don't have my git remotes properly configured in this case. Basically gitpod does not know I have a fork (which is somehow logically from git clone perspektive).
If, on the other hand, I launch my workspace from my fork, I cannot make use of pre-builds efficiently because my (master) branch is not in sync with the upstream branch.

See also https://community.gitpod.io/t/how-to-specify-users-fork-instead-of-main-repository/639/16

**Comment by frank-bee** [comment]
Ideas?
Maybe gitpod could  find-out that I have a private fork and ask me to configure the remotes "as if I would have cloned from the fork"?

**Comment by svenefftinge** [comment]
> Maybe gitpod could find-out that I have a private fork and ask me to configure the remotes "as if I would have cloned from the fork"?

I think that absolutely makes sense. But I fear our backlog is just too long atm so that we can commit to actually working on this, right now. If you start from your clone, the configuration should be as desired. And you can install a bot (e.g. https://github.com/marketplace/actions/fork-sync) that auto-syncs your fork with the remote one in order to get the prebuilds.

**Comment by frank-bee** [comment]
> We could add Fork command which lists accessible forks and switch to them? THen a user can start from upstream to get prebuild and run such command to switch?

That would be great. 
@svenefftinge I also thought about syncing my fork with some tooling. But the main reason I introduced gitpod in our hyperledger project, is to  have an easy way for new contributors to start with the development. Or For reviewers to easily work with PRs. 

> Right now one can do it from terminal as well. 

I think even a little simpler, the following might do the job as well:
```
git remote delete origin 
git remote add origin https://github.com/akosyakov/gitpod.git
```
Upstream will be automatically configured.
But this you have to configure for each workspace -> Button or even Popup or so in WS Code would be great

**Comment by AlCalzone** [comment]
I've got to agree here. As described in https://github.com/gitpod-io/gitpod/issues/5947, I maintain a repository which is heavily dependent on external contributors, some of which aren't very familiar with git and GitHub. For them, the workflow for working with forks in Gitpod is suboptimal to say the less.

Ideally, when they click on a link to open my repository in gitpod, their environment would be set up in a way that they can push to their fork and don't have to set it up. Although the Github extension helps a bit with creating a fork instead of pushing to my repo, this will fail the second time when the fork already exists and the extension doesn't switch to the new origin remote.
Additionally, the default `user:email` permission does not get you far, because the Github extension requires more permissions to log in.

I've helped myself for now with this startup script 
https://github.com/zwave-js/node-zwave-js/blob/d1e0de932d97046b9b6803348c2db334ae91707c/.gitpod/setup-git.sh
that asks the user some questions, guides though fork creation and sets up the remotes properly. However I think this should be automated in some way. Especially since Gitpod knows the user's name and yet it isn't available as an env variable, so the user must enter it again.
